lib/charliecloud.py

```diff
@@ -446,6 +447,12 @@ def set_(dst_key, *src_keys):
             except AttributeError:
                FATAL("can't parse config: bad Env line: %s" % line)
             self.metadata["env"][k] = v
+      # History.
+      try:
+         hist = config["history"]
+      except AttributeError:
+         hist = []
```

Comment:
1. Wouldn't this be `KeyError`? Regardless I don't personally like the "ask forgiveness" idiom of Python, preferring instead the "ask permission" model. The reasoning is that it can be tricky to know exactly where somewhat generic exceptions like `KeyError` come from, so you might be responding to a different problem than you expected (though not in this case I don't think).

2. Does the exception ever happen? Can we treat a config missing `history` key as malformed?